### PR TITLE
tests: enable CI testing of kraken images

### PR DESCRIPTION
**Pull request by andrewschoen**
Signed-off-by: Andrew Schoen <aschoen@redhat.com>

### CODE REVIEW

**Comment by andrewschoen** [comment]
jenkins test ceph_ansible2.1-kraken-centos7_cluster

**Comment by leseb** [comment]
I suspect you might need a bit of this PR to have a successful build on Kraken https://github.com/ceph/ceph-docker/pull/471

**Comment by andrewschoen** [comment]
@leseb yeah, I think you're right. I've merged #471 and rebased this PR.

File: tests/tox.sh [hunk 1]
```diff
@@ -36,11 +36,15 @@ fi
 # copy the files to the root for the
 # types of images we're going to build
 mkdir -p {base,daemon,demo}
-# starting with kraken, the base image does not exist
 cp -Lrv ceph-releases/$CEPH_RELEASE/$IMAGE_DISTRO/base/* base || true
 cp -Lrv ceph-releases/$CEPH_RELEASE/$IMAGE_DISTRO/daemon/* daemon
 cp -Lrv ceph-releases/$CEPH_RELEASE/$IMAGE_DISTRO/demo/* demo || true
 
+# remove the base dir any release after jewel
```
Comment (leseb) [inline tests/tox.sh]: I think build_imgs will recreate one. I'm pushing a fix for this.

Comment (leseb) [inline tests/tox.sh]: Can you rebase on top of master and perhaps remove this commit?
Thanks!

Comment (andrewschoen) [inline tests/tox.sh]: ok, I can try. the testing script is also creating these directories though.

https://github.com/ceph/ceph-docker/blob/master/tests/tox.sh#L38

Comment (leseb) [inline tests/tox.sh]: The check I added only build the image if base exists and not empty. In our case, it'll be empty so we're good.

**Comment by andrewschoen** [comment]
jenkins test ceph_ansible2.1-kraken-centos7_cluster

**Comment by andrewschoen** [comment]
jenkins test ceph_ansible2.1-kraken-xenial_cluster

**Comment by andrewschoen** [comment]
@leseb this looks like it might be a valid failure for kraken on centos7, thoughts?

https://jenkins.ceph.com/job/ceph-docker-prs-ceph_ansible2.1-kraken-centos7_cluster/8/console

**Comment by leseb** [comment]
jenkins test ceph_ansible2.1-kraken-xenial_cluster

**Comment by leseb** [comment]
jenkins test ceph_ansible2.1-kraken-centos7_cluster

**Comment by leseb** [comment]
jenkins test ceph_ansible2.1-kraken-centos7_cluster

**Comment by leseb** [comment]
jenkins test ceph_ansible2.1-kraken-xenial_cluster

**Comment by leseb** [comment]
jenkins test ceph_ansible2.1-kraken-centos7_cluster

**Comment by leseb** [comment]
jenkins test ceph_ansible2.1-kraken-xenial_cluster

**Comment by leseb** [comment]
jenkins test ceph_ansible2.1-kraken-centos7_cluster

**Comment by leseb** [comment]
jenkins test ceph_ansible2.1-kraken-centos7_cluster

**Comment by leseb** [comment]
jenkins test ceph_ansible2.1-kraken-xenial_cluster

**Comment by leseb** [comment]
jenkins test ceph_ansible2.1-kraken-centos7_cluster

**Comment by leseb** [comment]
jenkins test ceph_ansible2.1-kraken-xenial_cluster

**Comment by leseb** [comment]
jenkins test ceph_ansible2.1-kraken-xenial_cluster

**Comment by leseb** [comment]
jenkins test ceph_ansible2.1-kraken-centos7_cluster

**Comment by andrewschoen** [comment]
jenkins test this please

**Comment by andrewschoen** [comment]
I've changed this so we test agains ceph-ansible@master on PRs intead of ceph-ansible@stable-2.1.

**Comment by leseb** [comment]
@andrewschoen so now both jewel and kraken fail on stable-2.1?

**Comment by andrewschoen** [comment]
@leseb no, those jobs just got disabled so they show failing in this PR now.  They won't be tested on subsequent PRs.

I think it's smarter for us to test against ceph-ansible@master right now so we can move a bit quicker with fixes in ceph-ansible and not have to backport. We can schedule nightly tests against ceph-ansible@stable-2.1 if we'd like.

**Comment by leseb** [comment]
@andrewschoen I think yes it's still valuable to run tests against stable-2.1. We can link the results of those tests in the README. It'd be nice :)

**Pull request merged**